src/convert.jl

```diff
@@ -0,0 +1,228 @@
+module Conversion
+
+using ..SAM
+using ..BAM
+
+using BioSequences, BioAlignments
+
+global  const op_to_code = Dict(
+    'M' => 0x00,
+    'I' => 0x01,
+    'D' => 0x02,
+    'N' => 0x03,
+    'S' => 0x04,
+    'H' => 0x05,
+    'P' => 0x06,
+    '=' => 0x07,
+    'X' => 0x08,
+    'B' => 0x09,
+    '0' => 0x0a
+)
+
+function parse_cigar(record::SAM.Record)
+    
+    cigar_ops = ('M','I','D','N','S','H','P','=','X')
+    out = NamedTuple{(:L, :op), Tuple{Int64, Char}}[]
+
+    cigar = SAM.cigar(record)
+    start = 1
+
+    for (i,c) in enumerate(cigar)
+        if c ∈ cigar_ops
+            L = parse(Int, cigar[start:i-1])
```

Comment:
Avoid allocating a string for this
```suggestion
            L = parse(Int, view(cigar, start:i-1))
```